xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/plugin/rightsmanager/RightsManagerTest.java

```diff
@@ -0,0 +1,162 @@
+        String expectedValue = "XWiki.Some\\,OtherGroup,XWiki.GroupAfter";
+        // ThreadLocal: only to have this final and use it in the answer
+        final ThreadLocal<String> actualValue = new ThreadLocal<String>();
+
+        XWikiDocument rightsDocument = createMockDocument();
+
+        doReturn(Collections.singletonList(rightsDocument)).when(oldcore.getMockStore()).searchDocuments(any(), any(),
+            eq(oldcore.getXWikiContext()));
+
+        BaseObject rightsObject = Mockito.mock(BaseObject.class);
+        doReturn(Collections.singletonList(rightsObject)).when(rightsDocument)
+            .getXObjects(MockitoOldcore.GLOBAL_RIGHTS_CLASS);
+        doReturn(oldValue).when(rightsObject).getLargeStringValue("groups");
+        doAnswer(new Answer<Void>()
+        {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable
+            {
+                actualValue.set((String) invocation.getArgument(1));
+                return null;
+            }
+        }).when(rightsObject).setLargeStringValue(eq("groups"), any());
+
+        rights.replaceUserOrGroupFromAllRights(oldGroupName, newGroupName, false, oldcore.getXWikiContext());
+
+        assertEquals(expectedValue, actualValue.get());
+    }
+
+    @Test
+    public void testRemoveGroup() throws Exception
+    {
+        String oldValue = "XWiki.SomeGroup,XWiki.GroupToRemove,XWiki.Some\\,OtherGroup";
+        String expectedValue = "XWiki.SomeGroup,XWiki.Some\\,OtherGroup";
+        // ThreadLocal: only to have this final and use it in the answer
+        final ThreadLocal<String> actualValue = new ThreadLocal<String>();
+
+        XWikiDocument rightsDocument = createMockDocument();
+
+        doReturn(Collections.singletonList(rightsDocument)).when(oldcore.getMockStore()).searchDocuments(any(), any(),
+            eq(oldcore.getXWikiContext()));
+
+        BaseObject rightsObject = Mockito.mock(BaseObject.class);
+        doReturn(Collections.singletonList(rightsObject)).when(rightsDocument)
+            .getXObjects(MockitoOldcore.GLOBAL_RIGHTS_CLASS);
+        doReturn(oldValue).when(rightsObject).getLargeStringValue("groups");
+        doReturn("").when(rightsObject).getLargeStringValue("users");
+        doAnswer(new Answer<Void>()
+        {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable
+            {
+                actualValue.set((String) invocation.getArgument(1));
+                return null;
+            }
+        }).when(rightsObject).setLargeStringValue(eq("groups"), any());
+
+        rights.removeUserOrGroupFromAllRights("xwiki", "XWiki", "GroupToRemove", false, oldcore.getXWikiContext());
+
+        assertEquals(expectedValue, actualValue.get());
+    }
+
+    /**
+     * create a mock document that can be saved via the mock store
+     *
+     * @return the mocked document
+     */
+    private XWikiDocument createMockDocument()
```

Comment:
When using `MockitoOldcore` we tend to manipulate `XWikiDocument` instances as we usually would (get the document, modify it, save it, etc.) and not create any mock. It's mocking a memory storage of documents (but it's not interpreting queries, so you do need to mock `searchDocuments`).